dts/arm/nrf51_pca10028.dts

```diff
@@ -23,3 +23,42 @@
 	current-speed = <115200>;
 	status = "ok";
 };
+
+&flash0 {
+	/*
+	 * If chosen's zephyr,code-partition
+	 * is unset, the image will be linked
+	 * into the entire flash device.  If
+	 * it points to an individual
+	 * partition, the code will be linked
+	 * to, and restricted to that
+	 * partition.
+	 */
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x8000>;
+		};
+		slot0_partition: partition@8000 {
+			label = "image-0";
+			reg = <0x00008000 0x1a000>;
+		};
+		slot1_partition: partition@22000 {
+			label = "image-1";
+			reg = <0x00022000 0x1a000>;
+		};
+		scratch_partition: partition@3c000 {
+			label = "image-scratch";
+			reg = <0x0003c000 0x2000>;
```

Comment:
@d3zd3z @mbolivar Is this enough for scratch?

Reply:
My short answer is "too small for development purposes", but let me explain the tradeoffs so you can decide for yourself.

The key figures are:

- the ratio scratch size : image size
- the number of erase cycles supported by the flash hardware

With the scratch/image ratio of 1/13 given here, scratch will be erased 13 times every upgrade. Assuming flash sectors can support 10000 erase cycles without going out of spec (that's the case for nRF52, I'm assuming nRF51 is about the same), this configuration allows about 800 (10000 / 13 = 769) firmware upgrades before the device is out of spec.

That's enough for a product that's rarely updated, but for a developer board that's under heavy test, it would cut the lifetime quite a bit. I usually have a scratch/image ratio of 1/4 or 1/5, on advice from @d3zd3z.

Your call what to do with that info.

Reply:
@mbolivar thanks very much for the detailed explanation. I'd rather leave numbers that are useful to production and have devs tweak those during developments than have to deal with users complaining about slot size being too small to fit their apps

Reply:
I agree with this case.